ddprof-exporter/src/lib.rs

```diff
@@ -88,13 +89,19 @@ impl Request {
     async fn send(
         self,
         client: &HttpClient,
+        cancel: &CancellationToken,
     ) -> Result<hyper::Response<hyper::Body>, Box<dyn std::error::Error>> {
-        Ok(match self.timeout {
-            Some(t) => tokio::time::timeout(t, client.request(self.req))
-                .await
-                .map_err(|_| crate::errors::Error::OperationTimedOut)?,
-            None => client.request(self.req).await,
-        }?)
+        tokio::select! {
+            _ = cancel.cancelled() => Err(crate::errors::Error::UserRequestedCancellation.into()),
```

Comment:
One small optimisation I see is you could pass an `Option<&CancellationToken>` and do this, so that callers don't have to create a new token if none is provided

```suggestion
            _ = async { match cancel {
                Some(t) => t.cancelled().await,
                None => future::pending().await,
            }} => Err(crate::errors::Error::UserRequestedCancellation.into()),
```

Reply:
Fixed in abbf76c0ea20b55793481e481663d9236504c885